vendor/fasttext/fasttext.cc

```diff
@@ -43,8 +43,7 @@ void FastText::getVector(Vector& vec, const std::string& word) const {
 void FastText::saveVectors() {
   std::ofstream ofs(args_->output + ".vec");
   if (!ofs.is_open()) {
-    std::cerr << "Error opening file for saving vectors." << std::endl;
-    exit(EXIT_FAILURE);
+    throw std::runtime_error("Error opening file for saving vectors.");
```

Comment:
Why were all these changes necessary?

Reply:
It's so you can hit your `try`/`catch` block in the other file, right?

Reply:
Yes, and in case of error I didn't want the whole process to exit. The whole ruby or java process terminates quite abruptly.

Reply:
Seems like a good improvement.